em-app/src/lib.rs

```diff
@@ -174,3 +193,21 @@ fn get_user_data(pub_key: &Vec<u8>, config_id: Option<&str>) -> Result<[u8;64]>
 
     Ok(data)
 }
+
+pub fn get_certificate_expiry(cert: String) -> Result<DateTime> {
+    let cert = Certificate::from_pem(cert.as_bytes())
+        .map_err(|e| Error::CertificateFormat(e.to_string()))?;
+    let time = cert
+        .not_after()
+        .map_err(|e| Error::CertificateFormat(e.to_string()))?;
+
+    DateTime::new(
+        time.year().into(),
+        time.month().into(),
+        time.day().into(),
+        time.hour().into(),
+        time.minute().into(),
+        time.second().into(),
+    )
+    .ok_or_else(|| Error::CertificateFormat("No expiry date found".to_string()))
```

Comment:
Should this error message more about DateTime creation?
If there is No expiry date , this function should already failed at line 200.
```suggestion
    .ok_or_else(|| Error::CertificateFormat("Failed to parse expiry date into DateTime".to_string()))
```